src/utils.js

```diff
@@ -0,0 +1,217 @@
+
+  const windSpeed = document.createElement('p');
+  windSpeed.className = 'w-info';
+  windSpeed.innerText = wObject.windSpeed;
+  sideWrap.appendChild(windSpeed);
+
+  wrap.appendChild(sideWrap);
+
+  const addFav = document.createElement('button');
+  addFav.type = 'button';
+  addFav.id = 'add-fav';
+  addFav.className = 'add-fav fa fa-plus-square';
+
+  fetchDataFromLocalStorage('favorites').then(fav => {
+    if (fav.locations && fav.locations.find(loc => loc === wObject.location)) {
+      addFav.innerText = ' Remove from favorites';
+      addFav.addEventListener('click', (e) => {
+        e.stopPropagation();
+        removeFromFavorites();
+      });
+    } else {
+      addFav.innerText = ' Add to favorites';
+      addFav.addEventListener('click', (e) => {
+        e.stopPropagation();
+        addToFavorite();
+      });
+    }
+    wrap.appendChild(addFav);
+  });
+
+  wrap.style.display = 'block';
+
+  wrap.className = getImgSrc(wObject.cloudCond);
+
+  document.querySelector('.cel').addEventListener('click', (e) => {
+    e.stopPropagation();
+    e.preventDefault();
+    const tempStr = document.querySelector('.temp-fah');
+    if (tempStr) {
+      const tempVal = parseFloat(tempStr.innerText);
+      const cel = (((tempVal - 32) * 5) / 9).toFixed();
+      tempStr.className = 'temp-cel';
+      tempStr.innerText = cel;
+      document.querySelector('.fah').style.color = '#1a0dab';
+      document.querySelector('.cel').style.color = '#202124';
+    }
+  });
+
+  document.querySelector('.fah').addEventListener('click', (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    const tempStr = document.querySelector('.temp-cel');
+    if (tempStr) {
+      const tempVal = parseFloat(tempStr.innerText);
+      const fah = (tempVal * 1.8 + 32).toFixed();
+      tempStr.className = 'temp-fah';
+      tempStr.innerText = fah;
+      document.querySelector('.cel').style.color = '#1a0dab';
+      document.querySelector('.fah').style.color = '#202124';
+    }
+  });
+
+  document.getElementById('w-close-button').addEventListener('click', e => {
+    e.stopPropagation();
+    document.getElementById('w-wrap').style.display = 'none';
+  });
+
+  document.getElementById('notice').style.display = 'none';
+}
+
+
+export function createHeader() {
+  const headerWrap = document.createElement('div');
+  headerWrap.id = 'header-wrap';
+
+  const favLink = document.createElement('a');
+  favLink.className = 'favorites';
+  favLink.id = 'favorites';
+  favLink.href = '#';
+  favLink.innerHTML = 'Favorites <i class="fa fa-angle-double-right"></i>';
+  headerWrap.appendChild(favLink);
+
+  const favWrap = document.createElement('div');
+  favWrap.className = 'fav-wrap';
+  favWrap.id = 'fav-wrap';
+  favWrap.style.display = 'none';
+  headerWrap.appendChild(favWrap);
+
+  headerWrap.appendChild(showForm());
+
+  return headerWrap;
+}
+
+
+async function fetchJSON(url) {
+  return fetch(url).then(response => response.json());
```

Comment:
- [ ] It has error handling for promises or async/await.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png)

**Note** error handling here means that something went wrong with fetch itself (network error, server doesn't respond back, ...). You have already handled a missing city search, but we need also to check if any thing goes wrong with the fetch request.